assets/js/navigation.js

```diff
@@ -0,0 +1,10 @@
+var h2 = document.querySelectorAll("h2")
+var arrayLength = h2.length;
+var anchorList = document.getElementById("anchor-list");
+for (var i = 0; i < arrayLength; i++) { 
+	var li = document.createElement("LI");
+	var aTag = document.createElement("A");
+	aTag.innerHTML = '<a href="#' + h2[i].id + '">' + h2[i].textContent + '</a>';
```

Comment:
refactored code to not nest the anchor tags. Better result, less code, less anchors, much success.